File: Java/src/com/ecc/gkgo1a/Golem.java
package com.ecc.gkgo1a;

class Golem {
    int hp = 300;
    int defense = 80;
    int attack = 50;

    Golem() {}

    Golem(int hp)
    {
        this.hp = hp;
    }

    Golem(int hp, int defense)
    {
        this.hp = hp;
        this.defense = defense;
    }

    Golem(int hp, int defense, int attack) {
        this.hp = hp;
        this.defense = defense;
        this.attack = attack;
    }
}

class GolemEx {
    private final int GOLEM_HIT = 70;
    private final int GOLEM_FLEE = 20;
    public int Hp;
    public int Defense;
    public int Attack;
    public int Hit;
    public int Flee;

    GolemEx(int hp, int defense, int attack) {
        this.Hp = hp;
        this.Defense = defense;
        this.Attack = attack;
        this.Hit = GOLEM_HIT;
        this.Flee = GOLEM_FLEE;
    }
}

File: SwiftProject/SwiftProject/Golem.swift
//
//  Golem.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/05/05.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

let GOLEM_HIT = 70
let GOLEM_FLEE = 20

public struct Golem {
    var hp = 300
    var defense = 80
    var attack = 50
    
    init() {
        
    }
    
    init(hp: Int = 300, defense: Int = 80, attack: Int = 50) {
        self.hp = hp
        self.defense = defense
        self.attack = attack
    }
}

public struct GolemEx {
    var Hp: Int
    var Defense: Int
    var Attack: Int
    let Hit = GOLEM_HIT
    let Flee = GOLEM_FLEE
    
    init(hp: Int, defense: Int, attack: Int) {
        Hp = hp
        Defense = defense
        Attack = attack
    }
}
